### PR TITLE
Feature/hooks

### PR DESCRIPTION
**Pull request by AbnirHencazs**
### Features
1. Search functionality: 
Look for videos  using the search input/bar, available from navigation bar visible in all the application pages
2. Video detail view:
Now when you click in a video a new page (VideoPage) will render with and embedded iframe video which you can reproduce
3. Related videos feature:
Whenever you're on the VideoPage you'll see a list of related videos to the one you're watching.

## Important!

Since i've used all my quota i was put in the necesecity to deploy another build, please ignore the one indicated in the google form and instead refer to this: https://loving-leakey-1bad83.netlify.app/

### CODE REVIEW

**Comment by AbnirHencazs** [comment]
### Test Coverage screen-shot
Captura de pantalla (6)

**Comment by AbnirHencazs** [comment]
## Important! 

Since i've used all my quota i was put in the necesecity to deploy another build, please ignore the one indicated in the google form and instead refer to this: https://loving-leakey-1bad83.netlify.app/

File: src/pages/Home/Home.page.jsx [hunk 1]
```diff
@@ -1,12 +1,20 @@
 import React from 'react';
 import NavBar from '../../components/Navbar';
 import VideoCardList from '../../components/VideoCardList';
+import useGapi from '../../hooks/useGapi';
+import { useParams } from 'react-router-dom';
 
 function HomePage() {
+  
+  const { searchQuery } = useParams();
+  let { videos } = useGapi( searchQuery ? searchQuery : "", null)
```
Comment (erickwize) [inline src/pages/Home/Home.page.jsx]: If you're not changing the videos' variable, I recommend using `const` so we keep closer to a pure function

File: src/pages/Video/Video.page.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import VideoPlayer from '../../components/VideoPlayer';
+import RelatedVideos from '../../components/RelatedVideos'
+import Navbar from '../../components/Navbar';
+import { useHistory } from 'react-router-dom';
+
+const VideoPage = () => {
+    const history = useHistory()
+    const updateSearch = (childData) =>{
+        history.push(`/q:${childData}`)
```
Comment (erickwize) [inline src/pages/Video/Video.page.jsx]: The standard URL formatting is to use the value using paths or using query parameters, but you're having a combination of both. I recommend using `<path>?q=${childData}` or `<path>/q/{childData}`

File: src/components/VideoPlayer/index.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                                `${video.snippet.description.substring(0, 250)}... `
+                                :
+                                video.snippet.description
+                            }
+                        </p>
+                        
+                    </div>
+                    
+                    : "Sín titulo"
+                }
```
Comment (erickwize) [inline src/components/VideoPlayer/index.jsx]: It is not a good practice to have two ternary conditionals, it is hard to understand where the condition ends at first. I recommend doing something like:
```js
{ isLoading ? 'Cargando' : ''}
{ !isLoading && checkVideo(video) ? (...) : 'Sín título' }
```

File: src/components/App/App.component.jsx [hunk 4]
```diff
@@ -1,14 +1,33 @@
...
 
 function App() {
+
   return (
     <BrowserRouter>
       <Switch>
         <Route exact path="/">
           <HomePage />
         </Route>
+        <Route exact path="/q:searchQuery">
```
Comment (erickwize) [inline src/components/App/App.component.jsx]: Please read the recommendation on the URL standards below 😊

**Review by erickwize — APPROVED** [review]
## Comments
Nice work Brian! The UI is looking, good code practices and expected behaviors implemented. Nice testing and mock strategies. I liked the workers you implemented not only for testing purposes but for development (localhost) too.

I recommend adding a CSS property `cursor: pointer;` for the card so when the user hovers over it, the cursor changes.

## Acceptance Criteria
- [x] Videos in the Home View are fetched from the YouTube API and displayed correctly.
- [x] Users can search for YouTube videos using the search field on the header.
- [x] A video can be played from the **Video Details View** after clicking on it.
- [x] The video information and related videos list are displayed correctly on the **Video Details View**.
- [x] When a user clicks on a related video the video player, information and related videos list are updated.

### Bonus Points
- [x] Custom Hooks for API calls are implemented and tested correctly.
- [x] Testing coverage is above 60%. (Please include a screenshot of the code coverage report in your PR description).

File: src/pages/Video/Video.page.jsx (hunk 2 above)
Comment (AbnirHencazs) [reply src/pages/Video/Video.page.jsx]: Great! Thanks for the tip.

File: src/components/VideoPlayer/index.jsx (hunk 3 above)
Comment (AbnirHencazs) [reply src/components/VideoPlayer/index.jsx]: Great, thank you for the good practices recommendations.

File: src/pages/Home/Home.page.jsx (hunk 1 above)
Comment (AbnirHencazs) [reply src/pages/Home/Home.page.jsx]: Perfect, i was actually doubting about which reserved word to use, thank you for clarifying it.

**Pull request merged**